### DISCUSSION TITLE
Support global image loader config in next.config.js

### DISCUSSION THREAD

**Discussion by wong2** [vercel/next.js#24736] [Ideas]
### Describe the feature you'd like to request

Currently, the custom loader function must be passed to every next/image component, I'd like to add a global config in next.config.js

### Describe the solution you'd like

Support loader function config in next.config.js

```
images: {
  loader: ({ src, width, height }) => string
}
```

I'd like to implement this

**Comment by timneutkens** [comment]
You can create a component and use normal React composition:

```
import Image from 'next/image'
export default function MyImage(props) {
  return <Image loader={({src, width }) => `${src}?w=${width}`} {...props} />
}
```

**Reply by wong2** [reply]
I known, but this means I have to let everyone on my team learn to always use `MyImage` instead of `next/image`

**Reply by wong2** [reply]
With this method, `next export` would still give the error:

```
Error: Image Optimization using Next.js' default loader is not compatible with `next export`.
  Possible solutions:
    - Use `next start` to run a server, which includes the Image Optimization API.
    - Use any provider which supports Image Optimization (like Vercel).
    - Configure a third-party loader in `next.config.js`.
    - Use the `loader` prop for `next/image`.
  Read more: https://nextjs.org/docs/messages/export-image-api
```

Because it checks loader config in next.config.js https://github.com/vercel/next.js/blob/8d84b08752c5abbfcc90c34aa3b010253fd478c5/packages/next/export/index.ts#L340

**Reply by timneutkens** [reply]
`loader` can't be defined in `next.config.js` in any case because it's a dynamic function that needs to run in the browser as well.

**Comment by aiji42** [comment]
I had the same problem, so I solved it by creating this plugin [next-image-loader](https://github.com/aiji42/next-image-loader#city_sunrise-next-image-loader).
It transparently applies the custom loader defined in next.config.js to the props in `next/image`.

```js
// next.config.js
const withImageLoader = require('next-image-loader')(
  // (resolverProps: { src: string; width: number; quality?: number }) => string
  ({ src, width, quality }) => `${src}?width=${width}&quality=${quality || 75}`
)

module.export = withImageLoader({
  
})
```

Of course, I hope that official improvements will enable custom loaders to be transparently applied by defining them in one place such as next.config.js, without the need for plugins.
